sympy/concrete/tests/test_gosper.py

```diff
@@ -8,6 +8,8 @@
 def test_gosper_normal():
     assert gosper_normal(4*n + 5, 2*(4*n + 1)*(2*n + 3), n) == \
         (Poly(S(1)/4, n), Poly(n + S(3)/2), Poly(n + S(1)/4))
+    assert gosper_normal(4*n+5, 2*(4*n+1)*(2*n+3), n, polys=False) == \
+        (1/4, n + 3/2, n + 1/4)
```

Comment:
This will give Assertion error. LHS is simpified but the RHS is not. That is why it will result in an error. I think you should change this to `(S(1)/4, n + S(3)/2, n + S(1)/4)`

Reply:
Ah, you are right. Still it passes on everything but 2.7 and pypi. Thanks!

Reply:
Yes, I was concerned about that as well. Should this actually pass on Python 3?

Reply:
No idea why. I ran the tests on my (3.7) computer and didn't reflect on it (since it passed and I copied-and-pasted the example from the documentation).

Reply:
Same thing on my system as well. @asmeurer should this test pass on Python 3.7?

Reply:
> passes on everything but 2.7 and pypi

It seems to pass on my 2.7:
```
In [7]: gosper_normal(4*n+5, 2*(4*n+1)*(2*n+3), n, polys=False) == (S(1)/4, n + S(3)/2, n + S(1)/4)
Out[7]: True
```

Reply:
No, the problem is without using `S`. Without using `S` it is passing on 3.7 but not on 2.7. I think it should not pass on 3.7 as well

Reply:
I was maybe unclear in my description. It should have `S`, but it is surprising that it passes on 3.X without it.

Reply:
> I was maybe unclear in my description. It should have `S`, but it is surprising that it passes on 3.X without it.

Exactly

Reply:
That's because 

```py
>>> S(1)/4 == 1/4
True
```

Actually our float equality testing is a bit inexact (see https://github.com/sympy/sympy/issues/11707), but since 1/4 and 3/2 both have denominators that are powers of 2, the equality is exact (of course, even so, the test should test rational numbers, since that is what is actually output).